mux.go

```diff
@@ -107,6 +109,12 @@ func (mx *Mux) Options(pattern string, handlers ...interface{}) {
 	mx.handle(mOPTIONS, pattern, handlers...)
 }
 
+// NotFoundHandler sets a custom handler for the case when no routes matched
+// the given URL
+func (mx *Mux) NotFoundHandler(handler http.HandlerFunc) {
```

Comment:
I prefer `NotFound()` as well.. however, the handler arg should be a http.Handler, chi.Handler or interface{} where we can support various handler types. The other option is to have custom NotFound handlers as a middleware instead.